src/main/Java/SampleApp.java

```diff
@@ -0,0 +1,267 @@
+		System.out.println("b. Parse JSON response using com.google.gson");
+		System.out.println("c. Return token as string\n");
+		
+		System.out.println("Example 2: ");
+		System.out.println("a. Call the API /search using HTTP GET to get all Certified Product Details");
+		System.out.println("b. Parse the JSON using org.json to get only information keyed on \"product\"");
+		System.out.println("c. Parse the \"product\" JSON to CSV format");
+		System.out.println("d. Write the csv data to a file\n");
+		
+		System.out.println("Example 3: ");
+		System.out.println("a. call the API /search to get all Certified Product Details");
+		System.out.println("b. Save results to a .txt file\n");
+		
+		System.out.println("Example 4: ");
+		System.out.println("a. Call the API /data/search_options using token for authentication\n");
+		
+		System.out.println("NOTE: before running, please configure the environment.properties file\n");
+		
+		System.out.println("For example 1, please enter '1' into the console");
+		System.out.println("For example 2, please enter '2' into the console");
+		System.out.println("For example 3, please enter '3' into the console");
+		System.out.println("For example 4, please enter '4' into the console");
+		System.out.println("To perform all 4 examples, please enter '0' into the console");
+		System.out.println("To exit the application, please enter '9' into the console\n");
+	}
+	
+	/**
+	 * Returns a valid integer from the console
+	 * @return integer representing user input
+	 * @throws IOException for incorrect user console input
+	 */
+	public Integer getValidUserInput() throws IOException{
+		Scanner scanner = new Scanner(System.in);
+		Integer inputInt = scanner.nextInt();
+		if(inputInt.equals(null)){
+			System.out.println("User input integer is null. Setting to 0 as default.");
+			inputInt = 0;
+		}
+		else if(inputInt == 9){
+			System.out.println("Exiting application...");
+			System.exit(0);
+		}
+		else if(inputInt != 0 && inputInt != 1 && inputInt != 2 && inputInt != 3 && inputInt != 4){
+			System.out.println("Invalid user input. Please enter an integer between 0 and 4.");
+			System.exit(0);
+		}
+		scanner.close();
+		return inputInt;
+	}
+	
+	/**
+	 * Uses HTTP POST to call /auth/authenticate and get an authenticated token to be passed for future API calls that require an authenticated user.
+	 * Uses the Properties from environment.properties to obtain username, password, and API-key
+	 * Parses json response and returns the value of the token from the json
+	 * @return String with the token
+	 * @throws IOException for REST HTTP POST
+	 */
+	public String getToken() throws IOException {
+		System.out.println("\nRunning example 1 to get token for authentication:");
+		System.out.println("Making REST HTTP POST call to " + props.getProperty("targetHost") + props.getProperty("authenticate") + 
+				" using API-key=" + props.getProperty("apiKey").substring(0, 8) + "...");
```

Comment:
Oh, yeah, Dan asked me about this yesterday. I just didn't see much value in displaying the entire thing just for demo purposes. Maybe put a comment in about "truncating API-Key for better display" or something